recipes/microprofile/all/patches/cmake.patch

```diff
@@ -0,0 +1,141 @@
+--- /dev/null
++++ b/CMakeLists.txt
+@@ -0,0 +1,138 @@
++cmake_minimum_required(VERSION 3.4)
++project(microprofile)
++
++option(MP_ENABLED "Enable library" ON)
++option(MP_DEBUG "Enable debug mode" OFF)
++option(MP_MINIZ "Use Miniz library" OFF)
++option(MP_BIG_ENDIAN "Use big endian" OFF)
++option(MP_GPU_TIMER_CALLBACKS "Enable GPU timer callbacks" OFF)
++option(MP_GPU_TIMERS_GL "Enable OpenGL GPU timers" OFF)
++option(MP_GPU_TIMERS_D3D11 "Enable DirectX 11 GPU timers" OFF)
++option(MP_GPU_TIMERS_D3D12 "Enable DirectX 12 GPU timers" OFF)
++option(MP_GPU_TIMERS_VULKAN "Enable Vulkan GPU timers" OFF)
++
++set(MICROPROFILE_EXPORT_FILENAME microprofile.export.h)
++set(MICROPROFILE_PUBLIC_HEADERS
++        ${PROJECT_SOURCE_DIR}/microprofile.h
++        ${CMAKE_CURRENT_BINARY_DIR}/${MICROPROFILE_EXPORT_FILENAME}
++)
++
++set(THREADS_PREFER_PTHREAD_FLAG ON)
++find_package(Threads REQUIRED)
++
++add_definitions("-DMICROPROFILE_PER_THREAD_BUFFER_SIZE=${MP_THREAD_BUFFER_SIZE}")
++add_definitions("-DMICROPROFILE_PER_THREAD_GPU_BUFFER_SIZE=${MP_THREAD_GPU_BUFFER_SIZE}")
++add_definitions("-DMICROPROFILE_MAX_FRAME_HISTORY=${MP_MAX_FRAME_HISTORY}")
++add_definitions("-DMICROPROFILE_WEBSERVER_PORT=${MP_WEBSERVER_PORT}")
++add_definitions("-DMICROPROFILE_WEBSERVER_MAXFRAMES=${MP_WEBSERVER_MAXFRAMES}")
++add_definitions("-DMICROPROFILE_WEBSERVER_SOCKET_BUFFER_SIZE=${MP_WEBSERVER_SOCKET_BUFFER_SIZE}")
++add_definitions("-DMICROPROFILE_GPU_FRAME_DELAY=${MP_GPU_FRAME_DELAY}")
++add_definitions("-DMICROPROFILE_NAME_MAX_LEN=${MP_NAME_MAX_LENGTH}")
++add_definitions("-DMICROPROFILE_MAX_TIMERS=${MP_MAX_TIMERS}")
++add_definitions("-DMICROPROFILE_MAX_THREADS=${MP_MAX_THREADS}")
++add_definitions("-DMICROPROFILE_MAX_STRING=${MP_MAX_STRING_LENGTH}")
++add_definitions("-DMICROPROFILE_TIMELINE_MAX_TOKENS=${MP_TIMELINE_MAX_TOKENS}")
++add_definitions("-DMICROPROFILE_THREAD_LOG_FRAMES_REUSE=${MP_THREAD_LOG_FRAMES_REUSE}")
++add_definitions("-DMICROPROFILE_MAX_GROUPS=${MP_MAX_GROUPS}")
++
++if (MP_ENABLED)
++    add_definitions("-DMICROPROFILE_ENABLED=1")
++else()
++    add_definitions("-DMICROPROFILE_ENABLED=0")
++endif()
```

Comment:
FYI, when you do the following the defines are also visible on the exported target.
```suggestion
+target_compile_definitions("-DMICROPROFILE_ENABLED=$<BOOL:${MP_ENABLED}>")
```